action/src/plugins/imageDiff/plugin.ts

```diff
@@ -0,0 +1,80 @@
+import * as core from '@actions/core';
+import { randomUUID } from 'crypto';
+import fs from 'fs';
+import download from 'image-downloader';
+import { basename } from 'path';
+import pixelmatch from 'pixelmatch';
+import { PNG } from 'pngjs';
+
+import { ImageDiffPlugin } from '../../schemas/stoatConfigSchema';
+import { GithubActionRun, UploadImageDiffRequest } from '../../types';
+import { submitPartialConfig } from '../helpers';
+import { createSignedUrl, uploadPath } from '../staticHosting/helpers';
+
+const runImageDiffPlugin = async (
+  taskId: string,
+  taskConfig: ImageDiffPlugin,
+  { ghToken, ghRepository: { repo, owner }, ghSha }: GithubActionRun,
+  stoatConfigFileId: number
+) => {
+  core.info(`[${taskId}] Running static hosting plugin (stoat config ${stoatConfigFileId})`);
+  const currentDirectory = process.cwd();
+  core.info(`[${taskId}] Current directory: ${currentDirectory}`);
+
+  // get baseline image
+  if (!taskConfig.baseline) {
```

Comment:
`image_diff_plugin` in the JSON Schema doesn't require `baseline`.

Reply:
Right. My plan is to implement the default `baseline` in a separate PR.